src/cryptonote_basic/connection_context.h

```diff
@@ -73,6 +75,7 @@ namespace cryptonote
     int m_expect_response;
     uint64_t m_expect_height;
     size_t m_num_requested;
+    std::map<int, size_t> m_max_bytes;
```

Comment:
A sorted `std::vector` or a `boost::container::flat_map` would get better cache locality and require little changes. The values are currently constant, so they could be in a [fixed `static constexpr` array with a `std::lower_bound` lookup](L80).

If there isn't going to be unique values per-connection (eventually), the map should also be moved to some reusable location (or the constant version used) so the memory isn't duplicated. The config in levin handler actually make some sense - there's already a max packet value there, so the `net_node.inl` code can initialize this when it sets the "handler".

Reply:
It's very small so I kept to a linear search, simpler (for me).

Reply:
Are these values going to be per-connection modified? This can still be per-zone (at least) or even constant array.

And `std::lower_bound` provides the insert location of a sorted array (possible `end()`). It works with `std::pair` directly in this case, but is comparing the second element unnecessarily. Otherwise it requires a custom lambda which is a pita.

Reply:
Probably not per connection (though maybe for Tor/I2P since those don't get blocks so those can be set to 0), but possibly over time, to match expected block size. I guess I'll make it a binary search since you insist.